core/src/stores/postgres.rs

```diff
@@ -2357,11 +2345,14 @@ impl Store for PostgresStore {
         let total = match limit_offset {
             None => rows.len(),
             Some(_) => {
-                let count_sql = match table_id {
-                    Some(_) => "SELECT COUNT(*) FROM database_rows WHERE database_table = (SELECT id FROM databases_tables WHERE database = $1 AND table_id = $2)",
-                    None => "SELECT COUNT(*) FROM database_rows WHERE database_table IN (SELECT id FROM databases_tables WHERE database = $1)",
-                };
-                let t: i64 = c.query_one(count_sql, &params).await?.get(0);
+                let t: i64 = c
+                    .query_one(
+                        "SELECT COUNT(*) FROM databases_rows WHERE database_table = \
+                (SELECT id FROM databases_tables WHERE database = $1 AND table_id = $2)",
```

Comment:
Once you'll have queried the table first you can get rid of that SELECT.

On top of being aligned with the rest of the implementation it saves that select and saves the INNER JOIN